### DISCUSSION TITLE
Como de fato migrar para Front-end?

### DISCUSSION THREAD

**Discussion by manzoliric** [frontendbr/forum#1073] [Discussão]
Hey :)

Trabalho com programação a quase 1 ano e acabei de sair da minha primeira empresa, lá eu era registrado como "Programador Junior" e trabalhei com criação de Landing Pages, Blogs/Sites/E-commerce com Wordpress, criação de VPS, bom, eu vi de tudo um pouco.. (estou desempregado atualmente, pretendo estudar o máximo possível)

Depois de conhecer um pouco dos dois lados, quero arrumar um emprego como Front-end focado em Javascript e se for possível React, porém fui em duas entrevistas e não sei aonde errei, não tive feedback de nenhuma das duas entrevistas, o que me dificultou saber aonde eu errei, o que falta, o que preciso melhorar..

Agora a pergunta do tópico, como migrar de fato para o Front-end? pedi algumas opiniões e alguns disseram "HTML, CSS e JavaScript, Junior precisa saber isso" e não, não está assim.. o que eu realmente preciso aprender para arrumar um emprego e começar de fato nesse mundo do Front-end?

Montei um plano de estudos para tentar separar o que eu preciso estudar, alguém tem alguma dica?
![image](https://user-images.githubusercontent.com/26907629/39797714-c9fac866-5333-11e8-8262-d07efc41f4da.png)

Estou meio perdido.. listei apenas JavaScript e React nos estudos e queria saber se preciso me aprofundar em algo mais!

Abraços! 👍

**Comment by diessica** [comment]
linda a sua organização! está ótimo e é demais a visão geral que tu tá tendo! só duas coisas:
1) tu precisa construir muitos apps pra compensar a sua falta de experiência na área
2) vai levar muito mais tempo se tu tentar aprender react pela documentação

## para resolver 1
sugestões de apps para construir:
- https://github.com/tastejs/awesome-app-ideas
- https://github.com/learn-javascript-courses/student-projects/blob/master/app-ideas.md
- https://github.com/shovanch/fullstack-web-developer-path

não se sinta sobrecarregado, escolhe os que achar mais interessante, o importante é fazer! tenta construir do zero mesmo, sofrendo com o webpack – se for demais, usa o [create-react-app](https://github.com/facebook/create-react-app).  tu pode usar esses projetos como exemplos de coisas que trabalhou nas entrevistas.

## para resolver 2
os cursos do wesbos são bastante práticos:
- react: https://reactforbeginners.com/
- react com redux: https://learnredux.com/

outros cursos gratuitos: https://egghead.io/community/resources (se quiser pagar, o egghead vale muito a pena, tem muito conteúdo de react e redux!)

sugiro fazê-los com a documentação do react/redux lado a lado, sempre. isso vai te dar uma base legal que muita gente que trabalha todo dia com essas ferramentas não têm. assim como recomendo aprender javascript com a [MDN](https://developer.mozilla.org/en-US/) sempre como suporte – sem stack overflow, sem pedir ajuda no fórum de primeira.

## por fim
tu vai ter que ver bastante qual é o seu ritmo, se não conseguir pegar com redux e webpack, caso seja muito de uma vez só, vai ter que alguns passos pra trás: só react e webpack, ou só react... mas como tu está desempregado, sugiro de cara ir fazendo de tudo e entendendo no processo (no caso, tu não vai ter tempo de usar react o suficiente para ver qual é o use case do redux etc, como recomendam, por exemplo, mas esteja interessado em descobrir isso quando tiver tempo).

**Comment by diessica** [comment]
a propósito,

> "HTML, CSS e JavaScript, Junior precisa saber isso

quem te falou isso está certo sim! não que tu vai conseguir o emprego que quer com isso (realmente, provavelmente não), mas é problemático fazer front-end sem realmente entender HTML e CSS, com o ego de achar que é fácil. eu digo saber mesmo, escrever HTML acessível, semântico; fazer layouts mais complexos com CSS usando flexbox etc.

o output de toda essas abstrações que a gente têm estudado no final é ainda HTML, CSS e JS, é isso que importa para – e impacta – o usuário **(que é pra quem tu faz o produto)**. React é só o que tu decidiu usar para abstrair tudo, quem tá navegando na internet não tá nem aí pra isso! eu até meio que escrevi sobre isso uma vez: https://diessi.ca/blog/ui-components-that-abstract-mistakes/

**Comment by vitocaio** [comment]
Boa, cara na verdade é bem simples, procure aprender aquilo que as empresas pedem que são pertinentes ao Front-end, algumas empresas não sabem o que realmente devem pedir, por isso vemos SQL, Java em requisitos de vagas de Front, mas tudo que for sobre Front é bom ter uma ideia por cima, como voce ainda é Junior, não tem a obrigação de saber de fato e dominar esses assuntos, mas e importante saber pelo menos o Hello World rs

**Comment by Murillofilho86** [comment]
Excelente tópico @manzolirch, eu estou em uma situação muito parecida, a diferença é que eu ainda estou trabalhando... Mas estou muito querendo mudar de back pra front, tbm sou júnior...

@diessica muito bons esses links que você colocou de projetos para ganhar experiencia... Recentemente consegui contribuir para o meu primeiro projeto open source no  [awesome-br.github.io](https://github.com/awesome-br/awesome-br.github.io) foi uma coisa bem simples, mas me deixou empolgado, 
achei uma lista com projetos que vão do iniciante ao avançado... 

[https://github.com/training-center/divertidalista](url)

**Comment by manzoliric** [comment]
@diessica Obrigado por todas as dicas e indicações de ideias e cursos, achei incrível, dei uma parada por alguns dias mas agora voltei com tudo e pretendo estudar MUITO :hammer: 
Vou seguir esses passos, estudar de segunda a quinta e sexta-feira/sabado e domingo praticar o máximo possível do que aprendi na semana e botar em praticas em apps, para caso na entrevista perguntar se eu sei, terei apps criados!

E sobre o HTML, CSS e JS, eu concordo, eles são a base para tudo que for trabalhar, o resto é bibliotecas, pre processadores, frameworks, mas a base mesmo são eles..

Obrigado de novo pelas palavras e pelas dicas, vou seguir e espero que de tudo certo!

**Comment by leandroaps** [comment]
Um perfil de onde trabalho hoje é o de DevOps, que tende a ser uma
premissa, onde envolve ou front ou back-end com operações, apesar de ter
muita experiência de mercado, continuo estudando JS (ES6) e meu foco tem
sido hoje no React, por base, você deve ter um ótimo conhecimento de CSS3 e
HTML você deve conhecer bem e como DevOps, Docker, Kubernetes e Google
Cloud. Mercado de front hoje tem muita informação, e é preciso foco, não
adianta saber tudo e não se especializar em nada! Veja qual linguagem te
agrada mais e foco nela! :D

A base dos meus projetos hoje, giram em torno dessas tecnologias:

Nginx
Node
React
JS (ES6)
Docker
Kubernetes
Jest + Pupeteer
Jenkins
-- 

[]´s

*Leandro Ap. de Siqueira*
leandroaps@gmail.com
+55 19 991112239

**Comment by LucasLeandro1204** [comment]
Se você não tem nenhuma noção de componentes e tal, eu te recomendo começar aprendendo Vue. Essa série [aqui](https://laracasts.com/series/learn-vue-2-step-by-step) mostra bastante coisa do Laravel junto, mas o foco é Vue e eu acredito ser a melhor introdução que vc possa ter. Depois você já vai ter knowledge o suficiente pra aprender React ou Angular (aprende react) 10x mais rápido.

Faça projetos pequenos mas usando diversos recursos, por exemplo:
Um todo list, mas com uma boa UI / UX, usando um state management (vuex / redux), com sistema de rotas (vue-router / react-router) e testes (unitários, integração, etc).
